File: 946-Validate-Stack-Sequences/946-Validate-Stack-Sequences.java
class Solution {
    public boolean validateStackSequences(int[] pushed, int[] popped) {
        Stack<Integer> stack = new Stack<>();
        int p = 0;
        for(int i = 0; i < pushed.length; i++){
            stack.push(pushed[i]);
            while(!stack.isEmpty() && stack.peek() == popped[p]){
                stack.pop();
                p += 1;
            }
        }
        while(!stack.isEmpty()){
            if(stack.pop() != popped[p]){
                return false;
            }
            p += 1;
        }
        return true;
    }   
}

File: 946-Validate-Stack-Sequences/946-Validate-Stack-Sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        stack = []
        p = 0
        for i in pushed:
            stack.append(i)
            while stack and stack[-1] == popped[p]:
                stack.pop()
                p += 1
        print(stack)
        for i in popped[p:]:
            if i  != stack.pop():
                return False

        return True